### PR TITLE
boards: Add support for Blue Clover PLT Demo v2

### PR DESCRIPTION
**Pull request by JeremyBCD**
This PR adds support for the Blue Clover PLT Demo V2, a new open source ([OSHWA certified](https://certification.oshwa.org/us002054.html)) demonstration and development board.

The board features a Nordic Semiconductor nRF52832.

This support has been tested with [the hardware](https://bcdevices.com/products/plt-demo-board) against several Zephyr samples.

### CODE REVIEW

**Comment by JeremyBCD** [comment]
Revised blueclover_plt_demo_v2_nrf52832.dts and blueclover_plt_demo_v2_nrf52832.yaml to address sample build issues.

File: boards/arm/blueclover_plt_demo_v2_nrf52832/README.rst
Comment (gmarull) [inline boards/arm/blueclover_plt_demo_v2_nrf52832/README.rst:1]: please delete this README, the board already has docs.

File: boards/arm/blueclover_plt_demo_v2_nrf52832/doc/index.rst
Comment (gmarull) [inline boards/arm/blueclover_plt_demo_v2_nrf52832/doc/index.rst:39]: no tabs here, just start aligned with upper `n`

**Review by gmarull — COMMENTED** [review]
LGTM, two minor requests

File: boards/arm/blueclover_plt_demo_v2_nrf52832/Kconfig.board
Comment (mbolivar-nordic) [inline boards/arm/blueclover_plt_demo_v2_nrf52832/Kconfig.board:6]: Nit: there is an extraneous extra space here

**Review by mbolivar-nordic — COMMENTED** [review]
LGTM assuming @gmarull's comments are addressed.

**Comment by JeremyBCD** [comment]
@gmarull @mbolivar-nordic Thanks for your review notes! I think that I've addressed everything.

**Review by gmarull — APPROVED** [review]

File: boards/arm/blueclover_plt_demo_v2_nrf52832/blueclover_plt_demo_v2_nrf52832.dts
Comment (anangl) [inline boards/arm/blueclover_plt_demo_v2_nrf52832/blueclover_plt_demo_v2_nrf52832.dts:33]: Perhaps it would be better to use labels like "STATUS push button" and "RESET push button" for consistency with descriptions of these buttons on the PCB.

**Review by anangl — APPROVED** [review]

**Pull request merged**
